src/main/java/Duke.java

```diff
@@ -23,39 +23,75 @@ public static void main(String[] args) {
-                case "event":
-                    String[] event = arr[1].split("/at ", 2);
-                    list[index] = new Event(event[0], event[1]);
-                    System.out.println("Got it. I've added this task:\n" + list[index]);
-                    System.out.println("Now you have " + index + " tasks in the list");
-                    index++;
-                    break;
-                default:
-                    System.out.println("Invalid command");
+            try {
+                switch (command) {
+                    case "list":
+                        if (index == 1) {
+                            throw new DukeException("OOPS!!! You have no tasks to be displayed.");
+                        } else {
+                            System.out.println("Here are the tasks in your list:");
+                            for (int i = 1; i < index; i++) {
+                                System.out.println(i + "." + list[i]);
+                            }
+                        }
+                        break;
+                    case "done":
+                        if (arr.length == 1) {
+                            throw new DukeException("OOPS!!! Please enter a task number to check as done e.g done 1");
+                        } else {
+                            int num = Integer.parseInt(arr[1]);
+                            if (num >= index || num < 1) {
+                                throw new DukeException("This task does not exist.");
+                            } else {
+                                System.out.println(list[num].done());
+                            }
+                        }
+                        break;
+                    case "todo":
+                        if (arr.length == 1) {
+                            throw new DukeException("OOPS!!! The description of a todo cannot be empty.");
+                        } else {
+                            list[index] = new Task(arr[1]);
+                            System.out.println("Got it. I've added this task:\n" + list[index]);
+                            System.out.println("Now you have " + index + " tasks in the list");
+                            index++;
+                        }
+                        break;
+                    case "deadline":
+                        if (arr.length == 1) {
+                            throw new DukeException("OOPS!!! The description of a deadline cannot be empty.");
+                        } else {
+                            String[] deadline = arr[1].split("/by ", 2);
+                            if (deadline.length == 1) {
+                                throw new DukeException("OOPS!!! Please enter a due date. e.g complete homework /by 1 Jan");
+                            } else {
+                                list[index] = new Deadline(deadline[0], deadline[1]);
+                                System.out.println("Got it. I've added this task:\n" + list[index]);
```

Comment:
I really like how you split the different processing of the commands which makes it very easy to follow!!! :)